templates/_macros/rich-vertical-list.jinja

```diff
@@ -0,0 +1,127 @@
+# Params
+# title (string) (required): Title of the rich vertical list
+# flipped (boolean) (optional): Whether the list items are flipped so image is on the left and the text is on the right. Defaults to false.
+# Slots
+# description: Paragraph-style description content
+# logo_section_image_[1-8]: Logo section image, assumed to be img.p-logo-section__logo
+# list_item_[1-15]: List item content, assumed to be li.p-list__item
+# image (required)
+{% macro rich_vertical_list(
+  title,
+  flipped=false
+) -%}
+  {% set description_content = caller('description') %}
+  {% set has_description = description_content|trim|length > 0 %}
+  {% set has_logo_section = caller('logo_section_image_1')|trim|length > 0 %}
+  {% set cta_content = caller('cta') %}
+  {% set has_cta = cta_content|trim|length > 0 %}
+  {% set has_list = caller('list_item_1')|trim|length > 0 %}
+  {% set image_content = caller('image') %}
+  {% set max_list_items = 15 %}
+  {% set max_logo_section_images = 8 %}
+
+  {#-
+    Construct list of list items using caller in the top-level macro
+    The _text_column_contents macro will not have access to the caller block, so we need to extract the list items here.
+  -#}
+  {% set list_items = [] %}
+  {% if has_list %}
+    {% for list_item_index in range(1, max_list_items + 1) %}
+      {% set list_item_content = caller('list_item_' + list_item_index|string) %}
+      {% set has_list_item_content = list_item_content|trim|length > 0 %}
+      {% if has_list_item_content %}
+        {{ list_items.append(list_item_content) or ""}}
```

Comment:
This is needed due to a quirk with how Jinja works - `List::append()` returns `None`, so when it's called in Jinja, it will render `None` unless you use an `or ""` or use `{% set _ = list.append(...) %}`.

![Screenshot from 2024-08-16 17-09-50](https://github.com/user-attachments/assets/cda3a90b-f7ad-477c-bbb9-0697b5412264)

This is also in [hero](https://github.com/canonical/vanilla-framework/pull/5189#discussion_r1684643581) ([src](https://github.com/canonical/vanilla-framework/blob/154e8584179e4d6fca56745e3c7d03fbb9526db1/templates/_macros/hero.jinja#L68))